src/sentry/incidents/tasks.py

```diff
@@ -40,7 +40,7 @@ def handle_subscription_metrics_logger(
     from sentry.incidents.subscription_processor import SubscriptionProcessor
 
     try:
-        if subscription.snuba_query.dataset == Dataset.Metrics.value:
+        if subscription.snuba_query and subscription.snuba_query.dataset == Dataset.Metrics.value:
             processor = SubscriptionProcessor(subscription)
```

Comment:
this actually just got fixed by my other branch -- since snuba_query was actually never None